renku/core/dataset/dataset_add.py

```diff
@@ -155,6 +159,21 @@ def _download_files(
     **kwargs,
 ) -> List["DatasetAddMetadata"]:
     """Process file URLs for adding to a dataset."""
+    if dataset.storage and dataset.storage.lower().startswith("s3://"):
+        if len(urls) == 0:
+            raise errors.ParameterError("No URL is specified")
+        if len(urls) > 1:
+            raise errors.ParameterError("Can only add 1 uri with S3 datasets.")
```

Comment:
Is there a problem to have more than one URL? We can check if they belong to the same storage as the dataset in the `for` loop below.

Reply:
Yes if you mount s3 data then you cannot mount more than one uri to the same location.

If you copy stuff too it is a problem. What is there are conflicts/duplicates between the two uris? Which files will end up in the dataset? Which will be overwritten? But this is definitely more solvable if you are copying and not mounting.

Perhaps to keep things simple to start we enforce this? I am happy to remove it. You will have to deal with it in the PR on copying stuff.

Reply:
URIs comes from the user: For example, when a user runs `renku dataset add s3-data s3://bucket/path/to/file-1 s3://bucket/path/to/file-2`, we will receive `s3://bucket/path/to/file-1` and `s3://bucket/path/to/file-2` in this function. In this case, it's quite convenient to allow multiple URIs.
It's users' responsibility if they add files that take a long time to process; we can't/shouldn't prevent users from doing that. Also, AFAIK we never check for addition on duplicate data. It's possible to run `renku dataset add my-data /path/to/dir /path/to/dir/a/file/inside`. They won't conflict in such cases because they go in different sub-directories.

Reply:
Ok. I removed this limitation too.